cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -525,11 +527,10 @@ func (c *machineController) nodeGroupForNode(node *corev1.Node) (*nodegroup, err
 			if err != nil {
 				return nil, fmt.Errorf("failed to build nodegroup for node %q: %v", node.Name, err)
 			}
-			// We don't scale from 0 so nodes must belong
-			// to a nodegroup that has a scale size of at
-			// least 1.
-			if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
-				return nil, nil
+			if !nodegroup.scalableResource.CanScaleFromZero() {
```

Comment:
It seems to me the comment is misleading?
shouldn't this be always the cases regardless of CanScaleFromZero? max-min must be at least 1.
cc @frobware

Reply:
What happens, even today, if you set min:0, max:0? we get 0, but prior to this effort we had to have at least 1 node.

Reply:
mm I'm struggling to see how this relates to this PR/support for scaling from zero.
Before this PR you could  have `min:1, max:1` which would fail the check. Why would we change that for the case where `min:0`?
Seems to me `if nodegroup.MaxSize()-nodegroup.MinSize() < 1` is just preventing you from having the same max and min.

Reply:
i've changed this to be similar to the logic in the machinedeployments section, so basically only checking if MaxSize-MinSize < 1. hopefully i've understood the logic properly XD